interfaces/Config_semgrep.atd

```diff
@@ -26,6 +26,7 @@ type t = {
   ~taint_unify_mvars <ocaml default="false"> : bool;
   ~taint_assume_safe_functions <ocaml default="false"> : bool;
   ~taint_assume_safe_indexes <ocaml default="false"> : bool;
+  ~taint_builtin_propagators <ocaml default="true"> : bool;
```

Comment:
When I saw this, my first reaction was "why would we want to do this?". We're probably going to lose the knowledge of this. Can we include a note that secrets was the use case that prompted it? Also, maybe we can rename this from "builtin_propagators" to "only_propagate_through_assignments" or even "secrets_mode"

Reply:
Fair enough, I will add some docs here, and when it has been tested a bit also do semgrep.dev/docs. Regarding the name I'm less sure, but I could take `only_propagate_through_assignments` for now.